packages/ra-ui-materialui/src/button/PrevNextButton.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import {
+    useCreatePath,
+    useGetList,
+    useGetRecordId,
+    useListParams,
+    useResourceContext,
+    useTranslate,
+} from 'ra-core';
+import { NavigateBefore, NavigateNext } from '@mui/icons-material';
+import { Link } from 'react-router-dom';
+import { IconButton } from '@mui/material';
+
+export const PrevNextButton = (props: PrevNextButtonProps) => {
+    const { linkType = 'edit' } = props;
+    const translate = useTranslate();
+    const recordId = useGetRecordId();
+    const resource = useResourceContext();
+    const createPath = useCreatePath();
+    const [query] = useListParams({
+        resource,
+    });
+    const list = useGetList(resource, {
+        pagination: { page: query.page, perPage: query.perPage },
+        sort: { field: query.sort, order: query.order },
+        filter: { ...query.filter },
+    });
```

Comment:
you probably need to add a specific cache for this query (configurable via props, of course), to avoid that the `getList` call repeats on every page.